pegasus/helper_modules/multiple_extname_file_utils.rb

```diff
@@ -0,0 +1,41 @@
+module MultipleExtnameFileUtils
+  # Some utilities for dealing with files with multiple extension names (IE,
+  # "foo.tar.gz")
+
+  def self.all_extnames(filename)
+    # Grab an array representing all extensions applied to the
+    # given file.
+    # IE, "foo.erb.md" -> [".erb", ".md"]
+
+    # Use a positive lookahead regex rather than a simple string match so the
+    # periods will be included in the resulting array, to better match the
+    # functionality of File.extname
+    File.basename(filename).split(/(?=\.)/).drop(1)
+  end
+
+  def self.file_has_any_extnames(filename, extnames)
+    # Returns true if any of the extensions used by the given file are in the
+    # given list of extension names
+    !(extnames & all_extnames(filename)).empty?
+  end
+
+  def self.file_has_all_extnames(filename, extnames)
+    # Returns true if and only if all of the extensions used by the given file
+    # are in the given list of extension names
```

Comment:
Not sure if we're worried about vulnerabilities here, but we could add `uniq` to make sure the same extension hasn't been used twice?

Reply:
ooh, good catch

Reply:
Not sure if the method name ('file has _all_ extnames') matches the comment description of the expected behavior ('find all files [...] that use _only_ the given extnames). Should this be renamed to `file_has_only_extnames` or something else?